src/layouts/EditHomepage.jsx

```diff
@@ -145,17 +161,189 @@ export default class EditHomepage extends Component {
+          frontMatter: {
+            ...currState.frontMatter,
+            sections,
+          },
+        }));
+      } else {
+        // The field that changed is the dropdown placeholder title
+
+        this.setState((currState) => ({
+          ...currState,
+          frontMatter: {
+            ...currState.frontMatter,
+            sections: update(currState.frontMatter.sections, {
+              0: {
+                hero: {
+                  dropdown: {
+                    title: {
+                      $set: value,
+                    },
+                  },
+                },
+              },
+            }),
+          },
+        }));
       }
     } catch (err) {
       console.log(err);
     }
   }
 
+  deleteHeroDropdown = async () => {
+    try {
+      const { frontMatter } = this.state;
+      const newSections = update(frontMatter.sections, {
+        0: {
+          hero: {
+            dropdown: {
+              $set: undefined,
+            },
+            key_highlights: {
+              $set: [],
+            },
+          },
+        },
+      });
+
+      await this.setState((currState) => ({
+        ...currState,
+        frontMatter: {
+          ...currState.frontMatter,
+          sections: newSections,
+        },
+      }));
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  createHeroDropdown = async () => {
+    try {
+      const dropdownObj = DropdownConstructor();
+
+      const { frontMatter } = this.state;
+      const newSections = update(frontMatter.sections, {
+        0: {
+          hero: {
+            button: {
+              $set: undefined,
+            },
+            url: {
+              $set: undefined,
+            },
+            key_highlights: {
+              $set: undefined,
+            },
+            dropdown: {
+              $set: dropdownObj,
+            },
+          },
+        },
+      });
+
+      this.setState((currState) => ({
+        ...currState,
+        frontMatter: {
+          ...currState.frontMatter,
+          sections: newSections,
+        },
+      }));
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  deleteHeroDropdownElem = async (event) => {
+    try {
+      const { id } = event.target;
+
+      // Verify that the target id is of the format `dropdownelem-${dropdownsIndex}`
+      const idArray = id.split('-');
+      if (idArray[0] !== 'dropdownelem') throw new Error('');
+      const dropdownsIndex = parseInt(idArray[1], RADIX_PARSE_INT);
+
+      const { frontMatter } = this.state;
+      const newSections = update(frontMatter.sections, {
+        0: {
+          hero: {
+            dropdown: {
+              options: {
+                $splice: [[dropdownsIndex, 1]],
+              },
+            },
+          },
+        },
+      });
+
+      this.setState((currState) => ({
+        ...currState,
+        frontMatter: {
+          ...currState.frontMatter,
+          sections: newSections,
+        },
```

Comment:
Just a thought: would it be useful to save the state of the hero dropdown / hero button & key highlights separately so that they don't have to start from scratch? so like in `onFieldChange`

```
      this.setState((currState) => ({
        ...currState,
        frontMatter: {
          ...currState.frontMatter,
          sections: newSections,
        },
       heroDropdown: heroDropdownSection,
      }
```

Reply:
Good point - will list it as an issue for better UX

Reply:
Created https://github.com/isomerpages/isomercms-frontend/issues/18